FE/src/components/Dashboard.tsx

```diff
@@ -1,35 +1,49 @@
-import * as React from 'react';
+import { useEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
 import { fetchMovies } from '../redux/movieSlice';
-import image from '../styles/images/dashboard-bg.jpg';
-import Grid from '@mui/system/Unstable_Grid';
+import Grid from '@mui/material/Grid';
 import Movies from './Movies/Movies';
 import { useAppDispatch } from '../redux/reduxHooks';
 import { RootState } from '../redux/store';
+import { MovieType } from './Movies/MovieTypes';
+import { findMovies } from '../utils/movieMethods';
 
-const Dashboard = () => {
+type DashboardProps = {
+  searchedMovie: string,
+}
+
+const Dashboard = (props: DashboardProps) => {
+  const { searchedMovie } = props;
   const dispatch = useAppDispatch();
   const state = useSelector((state: RootState) => state.movies);
   const { movies } = state;
 
-  React.useEffect(() => {
+  const [filteredMovies, setFilteredMovies] = useState<MovieType[]>([]);
+
+  useEffect(() => {
     dispatch(fetchMovies());
   }, [dispatch]);
 
+  useEffect(() => {
+    const filtered = findMovies(searchedMovie, movies);
+    setFilteredMovies(filtered);
+  }, [searchedMovie, movies]);

```

Comment:
You could solve this with a useMemo instead of an internal state + useEffect combo